### PR TITLE
Added proxy support

### LINKED ISSUE #178: No proxy support for UIP users
For all the users who are using behind UIP from behind a proxy network, requests module used in scrape.py needs to be specified with a proxy configuration. This feature is not available.
There are multiple ways to do so:
1. Either we add params to the argparse while running UIP
2. We create a configuration file specifying the proxies. Then we add a tutorial for writing a config file for the user in the README. (Many orgs support this way)

### PR DESCRIPTION
**Pull request by hackrush01**
Added a method for retrieving proxy from system variables in scrape.py
Fixes #178

### CODE REVIEW

**Comment by hackrush01** [comment]
Can't understand how to make pass the test.
Tried coala in my pc still didn't work. but it passes pytest.
```
hackrush_st@hackrush:~/git-repos/UIP$ pytest
================================================================ test session starts =================================================================
platform linux -- Python 3.5.2+, pytest-3.0.4, py-1.4.31, pluggy-0.4.0
rootdir: /home/hackrush_st/git-repos/UIP, inifile: 
collected 5 items 

tests/test_scrape.py ....
tests/test_utils.py .

============================================================== 5 passed in 1.25 seconds ==============================================================
hackrush_st@hackrush:~/git-repos/UIP$
```

**Comment by abhsag24** [comment]
@hackrush01 `pytest` is the command for tests to run, `coala` is the command for coala

**Comment by hackrush01** [comment]
So what do I do?
I ran coala but it still didn't help. Why am I failing the travis-ci?

**Comment by abhsag24** [comment]
@hackrush01 no, 80 is PEP8 standard

**Comment by hackrush01** [comment]
So how do I correct my code?

**Comment by abhsag24** [comment]
@hackrush01 if its just the line length thats causing the error just split the code into multiple lines?

**Comment by hackrush01** [comment]
Then also it doesn't get corrected, I tried that with my earlier commit.

-Regards
-Amit Tulshyan

On 14-Dec-2016 3:02 pm, "Abhay Raizada" <notifications@github.com> wrote:

*@abhsag24* commented on this pull request.
------------------------------

In uiplib/scrape/scrape.py <https://github.com/NIT-dgp/UIP/pull/239>:

> @@ -10,7 +10,7 @@ def make_soup(url):
     '''
     makes soup, that is basically parsing the html document
     '''
-    response = requests.get(url, headers={'User-agent': 'UIP'})
+    response = requests.get(url, headers={'User-agent': 'UIP'},
proxies=getproxies())

if you're experiencing line-length difficulties

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/NIT-dgp/UIP/pull/239>, or mute the thread

**Comment by abhsag24** [comment]
i see no reason why it shouldnt get corrected did you try applying the suggested patch?

**Comment by hackrush01** [comment]
Suggested via coala, right?

**Comment by abhsag24** [comment]
yes

**Comment by hackrush01** [comment]
Tried it. But I'll try once more tonight.
See this
https://travis-ci.org/NIT-dgp/UIP/builds/183728191

**Comment by abhsag24** [comment]
because that is still longer than 80 as shown by the bear, you have gtmake it smaller than that. Can you solve this on your own please? its simple really

**Comment by hackrush01** [comment]
Okay, I'll do it by night.

**Comment by hackrush01** [comment]
Will this be merged?

File: uiplib/scrape/scrape.py [hunk 1]
```diff
@@ -64,7 +71,7 @@ def get_reddit_image_links(url, no_of_images):
     as link of the image scraped from reddit.
     """
     # reddit requires .json format for the URL
-    page = make_json(url+'/.json')
+    page = make_json(url + '/.json')
```
Comment (abhsag24) [inline uiplib/scrape/scrape.py]: Generally linting changes are in their own commits. But I guess this is fine

Comment (hackrush01) [inline uiplib/scrape/scrape.py]: Couldn't get you!

Comment (hackrush01) [inline uiplib/scrape/scrape.py]: I used autopep8

File: uiplib/scrape/scrape.py [hunk 2]
```diff
@@ -19,7 +22,10 @@ def make_json(url):
     """
     makes a dictionary out of a json file.
     """
-    response = requests.get(url, headers={'User-agent': 'UIP'})
+    response = requests.get(
+        url,
+        headers={'User-agent': 'UIP'},
+        proxies=getproxies())
```
Comment (abhsag24) [inline uiplib/scrape/scrape.py]: Can you add a comment explaining what this argument and the function called does?

Comment (hackrush01) [inline uiplib/scrape/scrape.py]: Yes.

File: uiplib/scrape/scrape.py (hunk 1 above)
Comment (abhsag24) [inline uiplib/scrape/scrape.py]: Changes in a commit need to be related. This change is not related to the commit which you made to solve an issue hence this change should be in an own commit

**Comment by abhsag24** [comment]
@hackrush01 I'm reviewing through my phone and I'm travelling till tomorrow afternoon. So I don't know how consistent I can be to get this merged very soon

**Comment by hackrush01** [comment]
Oh that's totally fine.

**Comment by abhsag24** [comment]
@hackrush01 did you check this for your college Proxies? is there a way to test this? can you add a test for it?

**Comment by hackrush01** [comment]
Yes, this works for college proxies. But I couldn't get you about that test thing! What is that?
This works with or without proxies(if that's what you mean).

**Comment by abhsag24** [comment]
@hackrush01 i meant' could you write tests for this? so that we know it'll work, see `test_scrape.py`  also you have a merge conflict
